docker/Dockerfile.linux-build

```diff
@@ -0,0 +1,22 @@
+FROM debian:bullseye-slim
```

Comment:
I'm not sure how much we should care, but this image really assumes an x86_64 docker host. e.g. when building this image on an m1 mac, the x86_64 compiler will not be available.

Reply:
@scareything Hrmm, true. Ideally the same container-based local build procedure would work for both arm64 and amd64 workstations acting as the Docker host. After I get this working on Linux we can debug any issues that present on your ARM Mac. I assume that Macs that are Docker hosts routinely run a Linux VM to run Linux containers.